samples/doc-snippets/src/main/java/com/example/application/ActivityAgentManager.java

```diff
@@ -0,0 +1,82 @@
+package com.example.application;
+
+import akka.Done;
+import akka.javasdk.annotations.ComponentId;
+import akka.javasdk.client.ComponentClient;
+import akka.javasdk.workflow.Workflow;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+
+// tag::all[]
+@ComponentId("activity-agent-manager")
+public class ActivityAgentManager extends Workflow<ActivityAgentManager.State> { // <1>
+  private static final Logger logger = LoggerFactory.getLogger(ActivityAgentManager.class);
+
+  public record State(String userQuery, String answer) { // <2>
+    State withAnswer(String a) {
+      return new State(userQuery, a);
+    }
+  }
+
+  private final ComponentClient componentClient;
+
+  public ActivityAgentManager(ComponentClient componentClient) { // <3>
+    this.componentClient = componentClient;
+  }
+
+  public Effect<Done> start(String query) { // <4>
+    return effects()
+        .updateState(new State(query, ""))
+        .transitionTo("activities")
+        .thenReply(Done.getInstance());
+  }
+
+  public Effect<String> getAnswer() { // <5>
+    if (currentState() == null || currentState().answer.isEmpty()) {
+      return effects().error("Workflow '" + commandContext().workflowId() + "' not started, or not completed");
+    } else {
+      return effects().reply(currentState().answer);
+    }
+  }
+
+  @Override
+  public WorkflowDef<State> definition() { // <6>
+    return workflow()
+        .addStep(suggestActivities())
+        .addStep(error())
+        .defaultStepRecoverStrategy(maxRetries(2).failoverTo("error"));
+  }
+
+  private Step suggestActivities() { // <7>
+    return step("activities")
+        .call(() -> {
+          return componentClient
+              .forAgent()
+              .inSession(sessionId())
+              .method(ActivityAgent::query)// <8>
+              .invoke(currentState().userQuery);
+        })
+        .andThen(String.class, suggestion -> {
+          logger.info("Activities: {}", suggestion);
+
+          return effects()
+              .updateState(currentState().withAnswer(suggestion)) // <9>
+              .end();
+        })
+        .timeout(Duration.ofSeconds(60));
+  }
+
+  private Step error() {
+    return step("error")
+        .call(() -> null)
+        .andThen(() -> effects().end());
```

Comment:
I think we should take the opportunity to see if can reduce boilerplate in workfows. For example, it would be nice with a step without a call.

(is above `null` result allowed?)